src/include/sof/audio/module_adapter/library/native_system_agent.h

```diff
@@ -0,0 +1,27 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * Author: Pawel Dobrowolski <pawelx.dobrowolski@intel.com>
+ */
+
+#ifndef __NATIVE_SYSTEM_AGENT__
+#define __NATIVE_SYSTEM_AGENT__
+
+#include <sof/audio/module_adapter/module/module_interface.h>
+#include <system_service.h>
+
+struct NativeSystemAgent {
+	AdspSystemService system_service;
+	uint32_t log_handle;
```

Comment:
Lets not use CamelCase naming, unless this is somehow a structure that is taken from a 3rd party header which cannot be modified.

Other than that we can always use `struct adsp_system_service`.

I think adding a small description of what are we trying to do and a small information about the design of the solution would greatly help the review.

Reply:
@mwasko do you have any docs for Native System Agent ?

Reply:
@lgirdwood  there is only documentation for IADK Modules Adapter, but it can be used as a reference:
https://thesofproject.github.io/latest/architectures/firmware/intel/ace/iadk_modules.html

Currently in SOF we support only IADK loadable modules. The code that @pjdobrowolski is implementing is to provide support for SOF loadable modules written in C that implement Module API interface directly.

Reply:
ah, got you - so this change allows module API loadable modules alongside IADK loadable modules.

Reply:
yes indeed. I've called it native because I had to came out with unique name, not just another sof_system_agent.

Reply:
@dbaluta I use existing structure with such naming. Do I have to refactor that structure in this PR or do it in seperate one?

Reply:
I think best to make the change and then we can merge.